### PR TITLE
LPS-101626 Remove Liferay.Util.isFunction

### PR DESCRIPTION
**Pull request by julien**
Here are the reasons why we are removing this method:

- It's basic JavaScript

- We already have `isFunction` provided by AUI (see `Lang.isFunction`),
  metal.js, as well as other inline definitions.

### CODE REVIEW

**Comment by julien** [comment]
ci:test:sf

**Comment by julien** [comment]
ci:test:relevant

File: modules/apps/calendar/calendar-web/src/main/resources/META-INF/resources/js/remote_services.js [hunk 1]
```diff
@@ -135,12 +135,12 @@ AUI.add(
 							return response.json();
 						})
 						.then(data => {
-							if (Liferay.Util.isFunction(callback.success)) {
+							if (typeof callback.success === 'function') {
```
Comment (wincent) [inline modules/apps/calendar/calendar-web/src/main/resources/META-INF/resources/js/remote_services.js:138]: I love how the inline code is actually shorter than the abstracted code. 😂

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js [hunk 2]
```diff
@@ -796,10 +796,6 @@
 			return viewable;
 		},
 
-		isFunction(val) {
-			return typeof val === 'function';
-		},
-
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js:802]: Same as I said on https://github.com/wincent/liferay-portal/pull/206 — I don't think we can actually remove this. See my comment here: https://github.com/wincent/liferay-portal/pull/206#pullrequestreview-394784893

We could certainly add a comment to methods like this:

```js
/**
 * @deprecated as of 7.3 — use `typeof val === 'function'` instead.
 */
```

(or whatever the format of such messages we usually use.) Should probably do the same on the other PR as well.

We could also consider adding something that prints deprecation messages on first use in `development` mode, but if anything, I think I would leave that for the end and do it all at once. (We might have something already for this that we could use or adapt or copy; I don't know.)

Comment (julien) [reply modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js:802]: > We could also consider adding something that prints deprecation messages on first use in development mode, but if anything, I think I would leave that for the end and do it all at once. (We might have something already for this that we could use or adapt or copy; I don't know.)

Funny you mention this, I was thinking about that too. We can talk about this when the time has come.

**Comment by julien** [comment]
Re-submitted [here](#209)

**Pull request closed without merging**